### PR TITLE
Add feature enabled metrics to metrics forwarder

### PR DESCRIPTION
**Pull request by khewonc**
### What does this PR do?

Add a feature enabled metric for all enabled features.

Metric format: `datadog.operator.<feature>.enabled`, e.g. `datadog.operator.ksm.feature.enabled`
Tags:
* `cr_name`: DDA name
* `cr_namespace`: DDA namespace

Value:
* `1.0` if feature enabled
* No metric sent if feature not enabled

### Motivation

What inspired you to submit this pull request?

### Additional Notes

Anything else we should know when reviewing?

### Describe your test plan

Write there any instructions and details you may have to test your PR.

### CODE REVIEW

**Comment by levan-m** [comment]
Regarding 
> No metric sent if feature not enabled

Can we derive percentages if we only record 'enabled' metrics? Does it make sense to record `0.0` as well?

File: controllers/datadogagent/controller_reconcile_v2.go [hunk 1]
```diff
@@ -95,6 +95,8 @@ func (r *Reconciler) reconcileInstanceV2(ctx context.Context, logger logr.Logger
 	newStatus := instance.Status.DeepCopy()
 
 	features, requiredComponents := feature.BuildFeatures(instance, reconcilerOptionsToFeatureOptions(&r.options, logger))
+	// update list of enabled features for metrics forwarder
+	r.updateMetricsForwardersFeatures(instance, features)
```
Comment (levan-m) [inline controllers/datadogagent/controller_reconcile_v2.go:99]: Should we do this towards the end of the reconcile function? If something is enabled and reconciler starts failing the metrics reported would be inaccurate.

**Review by CharlyF — APPROVED** [review]

File: controllers/datadogagent/controller_reconcile_v2.go (hunk 1 above)
Comment (CharlyF) [reply controllers/datadogagent/controller_reconcile_v2.go:99]: that's a good point - Though I think we could actually keep the submission here to reflect that the feature is enabled, but later on, improve the code to have the value be `0` if for some reason it errors out (either alone or with a global error). WDYT?

Comment (levan-m) [reply controllers/datadogagent/controller_reconcile_v2.go:99]: That's an interesting perspective, that way we would track intent of enabling feature and that's a reasonable thing to do. So, I could go either way.

**Pull request merged**